Include/internal/pycore_cpuinfo.h

```diff
@@ -0,0 +1,151 @@
+#ifndef Py_INTERNAL_CPUINFO_H
+#define Py_INTERNAL_CPUINFO_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef Py_BUILD_CORE
+#  error "this header requires Py_BUILD_CORE define"
+#endif
+
+#include <stdint.h>         // uint8_t
+
+/* Declare a member of 'py_cpuid_features' storing a CPUID bit. */
+#define _Py_CPUID_DECL_FEAT(X)    uint8_t X:1
+/* Declare a member of 'py_cpuid_features' storing a XCR0 bit. */
+#define _Py_CPUID_DECL_XCR0(X)    uint8_t X:1
```

Comment:
They seem to repeat it. Maybe they should follow DRY.

I think it may be changed to this
```
_Py_CPUID_DECL_REG_FLAG(X) uint8_t X:1
```
In this way, It can be clearly seen that this's a register.

Reply:
I'd like to have different semantics. `_Py_CPUID_DECL_FEAT` is really to indicate a feature bit obtained by CPUID using the corresponding instruction while `_Py_CPUID_DECL_XCR0` is to indicate a bit of the XCR0 register which is actually returned by `xgetbv` (they are different instructions but both are related to CPUID itself, hence the name). 

Otherwise, I'd agree to have a single macro. However, I wanted to highlight their semantic differences.

Reply:
But it seems that this's only a macro used for declaration, not getting. Maybe it's good to let the user know that this's used to declare a register flag bit.

Reply:
Ah I can change the name later though I think the comment is clear. Also I don't expect people to use it externally. This part of the code will probably remain internal (if it is even accepted). The declaration intent was the `DECL` part of the name which was pretty straightforward to me.

Reply:
You're right. In fact, after changing it's name, no user will define the register externally (maybe he really operates through `ioremap`, and the user can use the macro to determine that what he declares is a register flag without distinguishing the register name)

Reply:
Maybe the construction is misunderstood but the macros are just here to create uint8_t fields of bitsize 1. 

What the user will see (and use) is just 'flags' and they won't really care from where they were retrieved I think. However for me and probably future maintainers it's good to know the origin of the field and what it really describes.

Now, for simplicity I could just have a single macro and improve the comments at the structure level. Could be also better. Probably.

Reply:
I agree with your consideration for the future, so I think you can add a comment to explain its meaning.
```
Declare a member of 'py_cpuid_features' storing a register a flag bit.

Example of declaration some register
SSE: _Py_CPUID_DECL_REG_FLAG(sse)
AVX: _Py_CPUID_DECL_REG_FLAG(avx）
```